### PR TITLE
Treewide: abandon Google Code homepages

### PR DESCRIPTION
**Pull request by c0bw3b**
###### Motivation for this change
Part of the #30636 effort

This PR updates homepages of packages that were hosted on Google Code.

When necessary I also updated the source URL in order to use the Google API storage archive: `https://storage.googleapis.com/google-code-archive-downloads/v2/code.google.com/projectname/filename`

It returns the same file so hashes were not updated but it may still trigger some rebuilds.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by c0bw3b** [comment]
Don't merge yet, it needs some more work

**Comment by c0bw3b** [comment]
Ok not wip anymore.

I just completed successfully a `nox-review pr 30846` on my server running nixos-17.09-small :

```shell
Result in /run/user/1000/nox-review-1i83i6xc
total 0
lrwxrwxrwx 1 renaud users 56 28 oct.  01:37 result -> /nix/store/ygk4rak9hy7cig1501fqwzp73w7iz3a5-amarok-2.8.0
lrwxrwxrwx 1 renaud users 67 28 oct.  01:37 result-10 -> /nix/store/ppgazy2x6b8dxwhspd35x4w9wlannnaz-qtscriptgenerator-0.1.0
lrwxrwxrwx 1 renaud users 56 28 oct.  01:37 result-11 -> /nix/store/swfi0cjwxybk913fi167bs202haifxz8-arora-0.11.0
lrwxrwxrwx 1 renaud users 53 28 oct.  01:37 result-12 -> /nix/store/04m2rw2dh4fiaw6x6xzaw8y44s9l3bmj-kippo-0.8
lrwxrwxrwx 1 renaud users 57 28 oct.  01:37 result-13 -> /nix/store/1wzzdsvmp7fhx8bkxw8k64qyv2blj5hl-hyenae-0.36-1
lrwxrwxrwx 1 renaud users 56 28 oct.  01:37 result-2 -> /nix/store/ra78vja84m37zcjjkq80m27y0nghw6x4-udptunnel-19
lrwxrwxrwx 1 renaud users 65 28 oct.  01:37 result-3 -> /nix/store/pzb7rfxs6kmr60vrxrlrgjvwfl1hh5y5-open-vm-tools-10.1.10
lrwxrwxrwx 1 renaud users 68 28 oct.  01:37 result-4 -> /nix/store/977jh7plhs9fl0r5iiy79isjxn9p5rrp-fcitx-mozc-2.20.2673.102
lrwxrwxrwx 1 renaud users 58 28 oct.  01:37 result-5 -> /nix/store/0pyk5rz69939ig4hhhiniw5z7y0d54s9-gwt-java-2.4.0
lrwxrwxrwx 1 renaud users 53 28 oct.  01:37 result-6 -> /nix/store/93fvm0sya7xz1g25jxc9d0ckpq9yb9lz-daq-2.0.6
lrwxrwxrwx 1 renaud users 56 28 oct.  01:37 result-7 -> /nix/store/yiwpxg4v87bxxaq96v5g5iiivx5ibmfg-libdnet-1.12
lrwxrwxrwx 1 renaud users 57 28 oct.  01:37 result-8 -> /nix/store/ns31pqgrwfqdz62klqkr83rgic187djd-snort-2.9.8.2
lrwxrwxrwx 1 renaud users 65 28 oct.  01:37 result-9 -> /nix/store/y5r4c891dab83dqpg57hw8mg6a93pxwy-open-vm-tools-10.1.10
```

It took a little time, mostly in the build of kdelibs-4.14.36

**Comment by jtojnar** [comment]
Some websites (e.g. garmintools) do not look very official

**Comment by FRidh** [comment]
It would also be good to afterwards go through this list of packages and see which we want to keep. I already see some old and unmaintained packages which we should drop.

**Comment by c0bw3b** [comment]
Thanks for the feedbacks.

@jtojnar yes it's always a bit tricky when the former Gcode repo does not provide an explicit link to a new home. Two changes I made are debatable : `garmintools` and `gtkdialog` since they are more or less a third-party effort to clone the repo on GitHub in case the initial one should disappear.
I'm totally ok to keep the archived Gcode page as homepage for these two packages. We just don't know how long it will stay around... My guess is we will be deprecating those packages before Google decides to take it down.

@FRidh yes some of them are clearly outdated but I found myself reluctant to mark them as broken because as long as the source are available and it compiles, someone might still be using it (inside or outside of the nixpkgs tree).
The main goal of this PR was mostly about refreshing meta.homepage attributes and it would be up to the maintainers & users of each package to say what's still needed and what's not.

Two obvious cases though:
- `slimrat` : declared in a coma on its homepage and rapidshare does not exist anymore
- `tmin` : officially deprecated in favor of `afl-tmin` from the american fuzzy lop package already available in nixpkgs

-> Should I mark these 2 as broken ?

**Comment by c0bw3b** [comment]
Fixed the homepages for garmintools and gtkdialog (back to the archived GCode page).

slimrat and tmin are marked as broken.

**Comment by c0bw3b** [comment]
Concerns here?

**Review by orivej — APPROVED** [review]
@c0bw3b Thank you for finding the substitutes for all these URLs!

I've enabled the parallel building for `qtscriptgenerator` because `nox-review` was too slow.

File: pkgs/applications/networking/browsers/arora/default.nix
Comment (jtojnar) [inline pkgs/applications/networking/browsers/arora/default.nix]: This can be replaced by `fetchFromGitHub`.

**Comment by c0bw3b** [comment]
Oh yes good call! Indeed the nox-review I ran before took some time with kdelibs... 

There is still some more G.Code homepages in the repo but that is a good start I believe.
Thanks for your review on this @orivej

**Pull request merged**
